src/utils/ucc_component.c

```diff
@@ -74,25 +67,25 @@ ucc_status_t ucc_components_load(const char *framework_name,
     int    i, n_loaded, ifaces_array_size;
     char  *full_pattern, framework_pattern[UCC_MAX_FRAMEWORK_NAME_LEN + 16];
     ucc_status_t            status;
+    size_t                  pattern_size;
     ucc_component_iface_t **ifaces = NULL;
 
     framework->n_components = 0;
     framework->components   = NULL;
 
     if (strlen(framework_name) == 0 ||
         strlen(framework_name) > UCC_MAX_FRAMEWORK_NAME_LEN) {
-        //TODO ucc_error
-        fprintf(stderr, "unsupported framework_name length: %d",
-                strlen(framework_name));
+        ucc_error("unsupported framework_name length: %s, len %d",
+                  framework_name, strlen(framework_name));
         return UCC_ERR_INVALID_PARAM;
     }
     sprintf(framework_pattern, "ucc_%s_", framework_name);
 
-    full_pattern = (char *)ucc_malloc(strlen(ucc_global_config.component_path) +
-                                          strlen(framework_name) + 16,
-                                      "full_pattern");
+    pattern_size =
```

Comment:
UCX ususally use asprintf for similar tasks - does it make sense we use it too?

Reply:
asprintf is BSD extension. Looks like UCX uses ucm_asprintf which is internal implementation. Do you mean we should copy-paste that fn into UCC (we can replace the fn to actual ucm when it is exposed as public api) ?

Reply:
I seem to remember they did, probably changed since I last looked. anyway, I don't think it's that important, so probably fine for now.